atc/worker/client.go

```diff
@@ -156,6 +169,66 @@ func (client *client) CreateVolume(logger lager.Logger, volumeSpec VolumeSpec, w
 	return worker.CreateVolume(logger, volumeSpec, workerSpec.TeamID, volumeType)
 }
 
+var checkProcessSpec = runtime.ProcessSpec{
+	Path: "/opt/resource/check",
+}
+
+func (client *client) RunCheckStep(
+	ctx context.Context,
+	logger lager.Logger,
+	owner db.ContainerOwner,
+	containerSpec ContainerSpec,
+	workerSpec WorkerSpec,
+	strategy ContainerPlacementStrategy,
+	containerMetadata db.ContainerMetadata,
+	resourceTypes atc.VersionedResourceTypes,
+	timeout time.Duration,
+	checkable resource.Resource,
+) (result []atc.Version, err error) {
+	chosenWorker, err := client.pool.FindOrChooseWorkerForContainer(
+		ctx,
+		logger,
+		owner,
+		containerSpec,
+		workerSpec,
+		strategy,
+	)
+	if err != nil {
+		err = fmt.Errorf("find or choose worker for container: %w", err)
+		return
+	}
+
+	container, err := chosenWorker.FindOrCreateContainer(
+		ctx,
+		logger,
+		&NoopImageFetchingDelegate{},
+		owner,
+		containerMetadata,
+		containerSpec,
+		resourceTypes,
+	)
+	if err != nil {
+		err = fmt.Errorf("find or create container: %w", err)
+		return
```

Comment:
Why are we setting `err` to a new error here? We're throwing it away if we end up here aren't we since we return on the second line? Should we be using `logger` here?

Reply:
It uses named returns args, but none of the other implementations do this so maybe we should change this too @cirocosta?

Reply:
oh hahah yeaah, sorry for that - I just have a tendency of always using named return args 😅  will address that

Reply:
https://github.com/concourse/concourse/pull/5239/commits/4431ba83fa376ad1cf5cfb0b537f237ae7d34e54